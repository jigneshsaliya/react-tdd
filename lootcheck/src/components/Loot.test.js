import React from 'react';
import { mount, shallow } from 'enzyme';
import { Loot } from './Loot';

describe('Loot',()=> {
    
    let props = {balance: 10, bitcoin :{},fetchBitcoin:{}};
    let loot = shallow(<Loot {...props}/>);
    it('renders peroperly', () => {
        expect(loot).toMatchSnapshot();
    });

    describe('when mounted', ()=> {
        const mockFetchbitcoin = jest.fn();
        beforeEach(()=> {
            props.fetchBitcoin = mockFetchbitcoin;
            loot = mount(<Loot {...props}></Loot>)
        });

        it('dispatches the `fetchBitcoin()` method it receives it from props',()=> {
            expect(mockFetchbitcoin).toHaveBeenCalled();
        });

    });

    describe('wehn ther are valid bitcoin props',()=>{
        beforeEach(()=> {
            props = {balance: 10, bitcoin : {bpi: {USD: {rate: '1,000'}}}};
            loot = shallow(<Loot {...props}></Loot>)
        
        });

        it('displays proper bitcoin value',()=> {
            expect(loot.find('h3').text()).toEqual('Bitcoin balance: 0.01');
        });
    });
});