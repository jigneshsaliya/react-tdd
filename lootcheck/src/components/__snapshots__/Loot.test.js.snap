// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loot renders peroperly 1`] = `ShallowWrapper {}`;
